R/functions-utils.R

```diff
@@ -0,0 +1,49 @@
+#' Helper function to check the data type even if `x` is an `Rle`.
+#'
+#' @param x
+#'
+#' @param class2 `character(1)` specifying the class for which should be tested.
+#'
+#' @author Johannes Rainer
+#'
+#' @noRd
+.is_class <- function(x, class2) {
+    if (is(x, "Rle"))
+        is(x@values, class2)
+    else is(x, class2)
+}
+
+#' Helper to convert i to integer index for subsetting.
+#'
+#' @param i `character` `logical` or `integer` used in `[i]` for subsetting
+#'
+#' @param length `integer` representing the `length` of the object to be
+#'     subsetted
+#'
+#' @param names `character` with the names (rownames or similar) of the object
+#'
+#' @return `integer` with the indices
+#'
+#' @author Johannes Rainer
+#'
+#' @noRd
+.i_to_index <- function(i, length, names = NULL) {
```

Comment:
`MsCoreUtils`?

Reply:
Sure. I could add that. What do you think of a name `i2index`? I find `iToIndex` hard to read (and I'm not allowed to use snake case ;) )

Reply:
This one is IMHO pretty useful and should be part of `MsCoreUtils`. `i2index` is fine. I have a similar function(s) in `topdownr`: https://github.com/sgibb/topdownr/blob/master/R/utils.R#L468-L536 (slightly different implementation if you interested). There it is called `.subset`/`.subsetByCharacter`/`.subsetByNumeric`/... So you may also consider `subsetBy`.

Reply:
OK, I'll add it to `MsCoreUtils`.

Reply:
Added in PR https://github.com/rformassspectrometry/MsCoreUtils/pull/18